sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -19,12 +19,28 @@ public partial class TextAnalyticsClient
         private readonly Uri _baseUri;
         private readonly HttpPipeline _pipeline;
         private readonly ClientDiagnostics _clientDiagnostics;
-        private readonly string _subscriptionKey;
         private readonly string _apiVersion;
         private readonly TextAnalyticsClientOptions _options;
 
         private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
 
+        private string _subscriptionKey;
+
+        private string SubscriptionKey
```

Comment:
Let's think about naming on this.  It's a cross-cognitive concept, so we'll want to make sure it sounds nice and official.

Reply:
ServiceSubscriptionKey?

Reply:
Adding more name possibilities: SubscriptionKeyCredential, APIKeyCredential. 

Just noticing that the portal refers to the key as both subscription key and API key in single-service accounts, and only as subscription key for multi-service accounts.

Reply:
I like `SubscriptionKeyCredential` it helps to understand what to use it for, and as you point it out, works for both scenarios.

Reply:
I like that one too ... I've also spent more time thinking about the alternatives, including ApiSubscriptionKey, and I'm not as opposed to just SubscriptionKey as I initially was.  If there's a reason that we can't use the word Credential in something that's not in Azure.Identity, then SubscriptionKey may be the next best option.  I'll ask for Krzysztof's perspective.

Reply:
>use the word Credential in something that's not in Azure.Identity?

@schaabs

Reply:
adding @KrzysztofCwalina

Reply:
@schaabs, could to recommend a good name here?

Reply:
A couple of thoughts. If this type were to remain in this package I would say `TextAnalyticsSubscriptionKeyCredential`. While it's quite wordy it does make it quite clear what clients this credential applies to. If you wanted to shorten this you could use one of the other variant names the documentation gives this key, such as `TextAnalyticsApiCredential`, or even simply `TextAnalyticsKeyCredential`. 

However, reading over this [Cognitive Services Authentication Doc](https://docs.microsoft.com/en-us/azure/cognitive-services/authentication), it seems like this credential type would likely apply to subscription key authentication across all cognitive services. If this is the case I would suggest moving this to a shared assembly so the same credential type can be used across all the service clients in `Azure.AI` namespace. In this case I would call it replace the "TextAnalytics" portion of the names with "CognativeServices", giving `CognativeServicesSubscriptionKeyCredential`, `CognativeServicesApiKeyCredential`, or `CognativeServicesKeyCredential`. 

Or if you prefer "AI" instead, `AISubscriptionKeyCredential`, `AIApiKeyCredential`, or `AIKeyCredential`. I'm not sure of the casing here. I believe the guideline is two letter acronyms are supposed to be all uppercase. But in some fonts it kind of reads like "AL Key Credential".